### ISSUE TITLE
Error request context canceled

### ISSUE THREAD

**Issue by amitdrana** [aws/karpenter#1452]
Hi Team, 
 I am facing this issue while trying to deploy Karpenter via helm.
 Any idea..
```
ubuntu@ip-10-40-3-188:~/devops-preprod/karpenter$ kubectl logs karpenter-669bb94648-fb6ts -n karpenter controller
2022-03-02T13:44:37.416Z	INFO	Successfully created the logger.
2022-03-02T13:44:37.416Z	INFO	Logging level set to: debug
{"level":"info","ts":1646228677.5214024,"logger":"fallback","caller":"injection/injection.go:61","msg":"Starting informers..."}
2022-03-02T13:44:37.521Z	DEBUG	controller.aws	Using AWS region eu-west-1	{"commit": "fd19ba2"}
2022-03-02T13:44:37.521Z	DEBUG	controller.aws.launchtemplate	Hydrating the launch template cache with names matching "Karpenter-hyke-eks-cluster-preprod-*"	{"commit": "fd19ba2"}
panic: Unable to hydrate the AWS launch template cache, RequestCanceled: request context canceled
caused by: context canceled

goroutine 1 [running]:
github.com/aws/karpenter/pkg/cloudprovider/aws.(*LaunchTemplateProvider).hydrateCache(0xc00086d580, {0x27c0938, 0xc0003810b0})
	github.com/aws/karpenter/pkg/cloudprovider/aws/launchtemplate.go:234 +0x4f4
github.com/aws/karpenter/pkg/cloudprovider/aws.NewLaunchTemplateProvider({0x27c0938, 0xc0003810b0}, {0x28952a0, 0xc0003222d8}, 0xc00019caa0, 0xc0006dff80)
	github.com/aws/karpenter/pkg/cloudprovider/aws/launchtemplate.go:68 +0x1d1
github.com/aws/karpenter/pkg/cloudprovider/aws.NewCloudProvider({0x27c0938, 0xc000381080}, {0x1faaee0})
	github.com/aws/karpenter/pkg/cloudprovider/aws/cloudprovider.go:88 +0x4fb
github.com/aws/karpenter/pkg/cloudprovider/registry.newCloudProvider(...)
	github.com/aws/karpenter/pkg/cloudprovider/registry/aws.go:27
github.com/aws/karpenter/pkg/cloudprovider/registry.NewCloudProvider({0x27c0938, 0xc000381080}, {0x1faaee0})
	github.com/aws/karpenter/pkg/cloudprovider/registry/register.go:25 +0x28
main.main()
	github.com/aws/karpenter/cmd/controller/main.go:76 +0x257

```

**Comment by njtran** [comment]
Hey @amitdrana, thanks for opening the issue. Sorry you're running into this. Looks like you're running into some permissions issues. Can you do the following?

- Can you post the spec of your `karpenter/karpenter` serviceaccount? If you see the `eks.amazonaws.com/role-arn: arn:aws:iam::{ACCOUNT_ID}:role/{CLUSTER_NAME}-karpenter` line in there, you should make sure that you have an IAM role with that ARN in your account. 
- Did you create the cluster with the command in the `getting-started` guide?
- What version of eksctl are you using if so? `eksctl version` will give you the output.

**Comment by njtran** [comment]
I've seen this when creating fresh clusters with a newer version of `eksctl`. Creating a new cluster to see if using the `karpenter` spec for the `ClusterConfig` makes a difference.

**Comment by amitdrana** [comment]
thanks, @njtran  for the quick response. 
 you were right, the issue is due to the instance profile. giving the wrong name.:)

**Comment by amitdrana** [comment]
Also, I have on more requirements, 
I need to set Label on nodes in karpenter provisioned but not able to do so.
Label :
` node.kubernetes.io/role`

**Comment by amitdrana** [comment]
Also , facing reconcile error

`ERROR	controller.controller.termination	Reconciler error	{"commit": "82ea63b", "reconciler group": "", "reconciler kind": "Node", "name": "ip-10-40-17-119.eu-west-1.compute.internal", "namespace": "", "error": "terminating node ip-10-40-17-119.eu-west-1.compute.internal, terminating cloudprovider instance, terminating instance ip-10-40-17-119.eu-west-1.compute.internal, RequestError: send request failed\ncaused by: Post \"https://ec2.eu-west-1.amazonaws.com/\": dial tcp: lookup ec2.eu-west-1.amazonaws.com on 172.20.0.10:53: read udp 10.40.13.98:44550->172.20.0.10:53: i/o timeout"}
`

**Comment by ellistarn** [comment]
Is this potentially what you're running into? https://aws.github.io/aws-eks-best-practices/karpenter/#amazon-eks-private-cluster-without-outbound-internet-access

**Comment by nikoul** [comment]
If that helps anyone, I got this issue on my new EKS cluster because coredns was in a bad state.
Fixing coredns fixed karpenter.

**Comment by triceras** [comment]
> If that helps anyone, I got this issue on my new EKS cluster because coredns was in a bad state. Fixing coredns fixed karpenter.

I am curious to know did you fix coredns as I am facing similar issue.

**Comment by nikoul** [comment]
> > If that helps anyone, I got this issue on my new EKS cluster because coredns was in a bad state. Fixing coredns fixed karpenter.
> 
> @nikoul I am curious to know did you fix coredns as I am facing similar issue.

I had some taints and labels on my nodes that weren't explicitly allowed in the coredns deployment. That's why coredns pods were always pending. So I used node affinity and tolerations.

**Comment by raidlman** [comment]
I had a similar issue like @nikoul . But for me it was the eks-pod-identity-agent. When I added a taint to my nodes where karpenter is running on, I forgot to add the corresponding toleration to the eks-pod-identity-agent addon.

The thing was, that I didn't notice it right away because the pods continued to run on those machines. Only when the spot instances were replaced by other nodes, those new nodes were excluded by the daemonset.
